pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstanceRunnable.java

```diff
@@ -425,23 +432,33 @@ private void processResult(Record srcRecord,
     }
 
     private void sendOutputMessage(Record srcRecord, Object output) {
+        if (!(this.sink instanceof PulsarSink)) {
+            Thread.currentThread().setContextClassLoader(functionClassLoader);
```

Comment:
have you benchmarked this?  since this is done per message basis.  I see this has been used in quite a lot of places.

Reply:
Yes I have benchmarked setting  and getting the context class loader.  It takes negligible time (in the order on tens of nano seconds).  If you look that the implementation, the code is only setting a pointer or returning a pointer